docs/infrastructure/fleet-on-kubernetes.md

```diff
@@ -0,0 +1,132 @@
+# Deploying Fleet on Kubernetes
+
+In this guide, we're doing to install Fleet and all of it's application dependencies on a Kubernetes cluster. Kubernetes is a container orchestration tool that was open sourced by Google in 2014.
+
+## Installing Infrastructure Dependencies
+
+For the sake of this tutorial, we will use Helm, the Kubernetes Package Manager, to install MySQL and Redis. If you would like to use Helm and you've never used it before, you must run the following to initialize your cluster:
+
+```
+helm init
+```
+
+### MySQL
+
+The MySQL that we will use for this tutorial is not replicated and it is not Highly Available. If you're deploying Fleet on a Kubernetes managed by a cloud provider (GCP, Azure, AWS, etc), I suggest using their MySQL product if possible as running HA MySQL in Kubernetes can be difficult. To make this tutorial cloud provider agnostic however, we will use a non-replicated instance of MySQL.
+
+To install MySQL from Helm, run the following command. Note that there are some options that are specified. These options basically just enumerate that:
+
+- There should be a `kolide` database created
+- The default user's username should be `kolide`
+
+```
+helm install \
+  --name fleet-database \
+	--set mysqlUser=kolide,mysqlDatabase=kolide \
+  stable/mysql
+```
+
+This helm package will create a Kubernetes `Service` which exposes the MySQL server to the rest of the cluster on the following DNS address:
+
+```
+fleet-database-mysql:3306
+```
+
+We will use this address when we configure the Kubernetes deployment and database migration job, but if you're not using a Helm-installed MySQL in your deployment, you'll have to change this in your Kubernetes config files.
+
+#### Database Migrations
+
+The last step is to run the Fleet database migrations on your new MySQL server. To do this, run the following:
+
+```
+kubectl create -f ./examples/kubernetes/fleet-migrations.yml
```

Comment:
The migration job and the fleet deployment must share the same container image source (otherwise what migrations are you running?) The notes need to mention the image in the job. 

Also, another thing worth noting is that `replace -f` would delete and create the new job in a single command.